### ISSUE TITLE
[BUG] TableServiceClientBuilder.buildAsyncClient() disobeyes `azureNamedKeyCredential`

### ISSUE THREAD

**Issue by ohl-nemeses** [Azure/azure-sdk-for-java#22384]
**Describe the bug**
TableServiceClientBuilder does not obey `azureNamedKeyCredential`  is no connection string is set.

IMHO https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClientBuilder.java#L112 should be:
```
AzureNamedKeyCredential namedKeyCredential = azureNamedKeyCredential;
```

***Exception or Stack Trace***
```
java.lang.IllegalStateException: A form of authentication is required to create a client. Use a builder's 'credential()', 'sasToken()' or 'connectionString()' methods to set a form of authentication.
        at com.azure.data.tables.BuilderHelper.buildPipeline(BuilderHelper.java:97)
        at com.azure.data.tables.TableServiceClientBuilder.buildAsyncClient(TableServiceClientBuilder.java:158)
        at com.hybris.cloud.automation.siemforwarder.inbound.AssignmentHandler.retrieveAll(AssignmentHandler.java:52)
        at com.hybris.cloud.automation.siemforwarder.inbound.StorageAccountSiemMessageSource.retrieveMessages(StorageAccountSiemMessageSource.java:34)
        at reactor.core.publisher.FluxFlatMap$FlatMapMain.onNext(FluxFlatMap.java:386)
        at reactor.core.publisher.FluxConcatMap$ConcatMapImmediate.innerNext(FluxConcatMap.java:281)
        at reactor.core.publisher.FluxConcatMap$ConcatMapInner.onNext(FluxConcatMap.java:860)
        at reactor.core.publisher.FluxFlattenIterable$FlattenIterableSubscriber.drainAsync(FluxFlattenIterable.java:421)
        at reactor.core.publisher.FluxFlattenIterable$FlattenIterableSubscriber.drain(FluxFlattenIterable.java:686)
        at reactor.core.publisher.FluxFlattenIterable$FlattenIterableSubscriber.onNext(FluxFlattenIterable.java:250)
        at reactor.core.publisher.FluxSubscribeOnCallable$CallableSubscribeOnSubscription.run(FluxSubscribeOnCallable.java:251)
        at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:68)
        at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:28)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

**To Reproduce**
```
new TableServiceClientBuilder()
            .credential(new AzureNamedKeyCredential(accountName, accountKey))
            .endpoint(tableEndpoint)
            .buildAsyncClient();
```
fails... while as
```
new TableServiceClientBuilder()
            .connectionString("AccountName="+accountName+";AccountKey="+accountKey)
            .endpoint(tableEndpoint)
            .buildAsyncClient();
```
works nicely.

***Code Snippet***
see above.

**Expected behavior**
`TableServiceAsyncClient` created. No exception thrown.

**Screenshots**
Not relevant.

**Setup (please complete the following information):**
 - OS: Windows 10.
 - IDE: eclipse
 - Library/Libraries: com.azure:azure-data-tables:12.0.0
 
**Additional context**
None.

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by alzimmermsft** [comment]
@vcolin7 could you take a look at this

**Comment by vcolin7** [comment]
Hi @ohl-nemeses, thank you for reporting this issue. I have [an open pull request](https://github.com/Azure/azure-sdk-for-java/pull/22438) with a fix and will ship it with the next version of Azure Data Tables early next month.

**Comment by ohl-nemeses** [comment]
Thx!

**Comment by ohl-nemeses** [comment]
Fixed in `12.1.0`
